### PR TITLE
update changelog for 0.55.2

### PR DESCRIPTION
**Pull request by esc**
As title.

### CODE REVIEW

**Review by gmarkall — APPROVED** [review]
This looks fine on RTD.

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,3 +1,10 @@
+Version 0.55.2 (19 May, 2022)
+-----------------------------
+
+This is a maintenance release to support NumPy 1.22 and Apple M1.
+
+* PR `#8067 <https://github.com/numba/numba/pull/8067>`_: Backport #8027: Support for NumPy 1.22 (`stuartarchibald <https://github.com/stuartarchibald>`_)
```
[the remark this answers is not in the document]
Comment (gmarkall) [reply CHANGE_LOG:6]: Yeah, I approved this with the  assumption that that was the only PR that would need to go in and it was just going to churn through to completion.

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,3 +1,33 @@
+Version 0.55.2 (19 May, 2022)
+-----------------------------
+
+This is a maintenance release to support NumPy 1.22 and Apple M1.
+
+*Pull-Requests:
```
Comment (stuartarchibald) [inline CHANGE_LOG:6]:
```suggestion
Pull-Requests:
```

**Review by sklam — APPROVED** [review]
Checked all listed PRs in the checklist (#8066) are here.

File: CHANGE_LOG (hunk 2 above)
Comment (sklam) [inline CHANGE_LOG:6]:
```suggestion
Pull-Requests:
```

```
../CHANGE_LOG:6:Inline emphasis start-string without end-string.
```

**Comment by esc** [comment]
@stuartarchibald @sklam comments addressed.

**Pull request merged**